pkg/payload/payload.go

```diff
@@ -353,52 +358,49 @@ func getPayloadTasks(releaseDir, cvoDir, releaseImage, clusterProfile string) []
 	}}
 }
 
-func loadReleaseFromMetadata(releaseDir string) (configv1.Release, string, error) {
+func loadReleaseMetadata(releaseDir string) (configv1.Release, error) {
 	var release configv1.Release
 	path := filepath.Join(releaseDir, cincinnatiJSONFile)
 	data, err := os.ReadFile(path)
 	if err != nil {
-		return release, "", err
+		return release, err
 	}
 
 	var metadata metadata
 	if err := json.Unmarshal(data, &metadata); err != nil {
-		return release, "", fmt.Errorf("unmarshal Cincinnati metadata: %w", err)
+		return release, fmt.Errorf("unmarshal Cincinnati metadata: %w", err)
 	}
 
 	if metadata.Kind != "cincinnati-metadata-v0" {
-		return release, "", fmt.Errorf("unrecognized Cincinnati metadata kind %q", metadata.Kind)
+		return release, fmt.Errorf("unrecognized Cincinnati metadata kind %q", metadata.Kind)
 	}
 
 	if metadata.Version == "" {
-		return release, "", errors.New("missing required Cincinnati metadata version")
+		return release, errors.New("missing required Cincinnati metadata version")
 	}
 
 	if _, err := semver.Parse(metadata.Version); err != nil {
-		return release, "", fmt.Errorf("Cincinnati metadata version %q is not a valid semantic version: %v", metadata.Version, err)
+		return release, fmt.Errorf("Cincinnati metadata version %q is not a valid semantic version: %v", metadata.Version, err)
 	}
 
 	release.Version = metadata.Version
 
-	var arch string
-	if archInterface, ok := metadata.Metadata["release.openshift.io/architecture"]; ok {
-		if archString, ok := archInterface.(string); ok {
-			if archString == releaseMultiArchID {
-				release.Architecture = configv1.ClusterVersionArchitectureMulti
-				arch = string(release.Architecture)
-			} else {
-				return release, "", fmt.Errorf("Architecture from %s (%s) contains invalid value: %q. Valid value is %q.",
-					cincinnatiJSONFile, release.Version, archString, releaseMultiArchID)
-			}
-			klog.V(2).Infof("Architecture from %s (%s) is multi: %q", cincinnatiJSONFile, release.Version, archString)
-		} else {
-			return release, "", fmt.Errorf("Architecture from %s (%s) is not a string: %v",
-				cincinnatiJSONFile, release.Version, archInterface)
+	if archRaw, hasArch := metadata.Metadata["release.openshift.io/architecture"]; hasArch {
+		arch, isString := archRaw.(string)
+		if !isString {
+			return release, fmt.Errorf("Architecture from %s (%s) is not a string: %v",
+				cincinnatiJSONFile, release.Version, archRaw)
 		}
-	} else {
-		arch = runtime.GOARCH
-		klog.V(2).Infof("Architecture from %s (%s) retrieved from runtime: %q", cincinnatiJSONFile, release.Version, arch)
+
+		if arch != releaseMultiArchID {
+			return release, fmt.Errorf("Architecture from %s (%s) contains invalid value: %q. Valid value is %q.",
+				cincinnatiJSONFile, release.Version, arch, releaseMultiArchID)
+		}
+
+		release.Architecture = configv1.ClusterVersionArchitectureMulti
+		klog.V(2).Infof("Architecture from %s (%s) is multi: %q", cincinnatiJSONFile, release.Version, arch)
```

Comment:
The "metadata does not have architecture key, lets read runtime" part moved one layer above to `loadPayloadMetadata`

Otherwise the logic is identical, just was refactored into guard clause sequence.